consensus_decentralization/metrics/gini.py

```diff
@@ -27,4 +27,11 @@ def gini(array):
     array = np.sort(array)
     index = np.arange(1, array.shape[0] + 1)
     n = array.shape[0]
-    return (np.sum((2 * index - n - 1) * array)) / (n * np.sum(array))
+    # Normalize the array to prevent overflow
```

Comment:
I think these changes are fine, but there are two tests that fail because of floating point errors that are produced here. One option is to update the tests to use the new values, as they are not exactly wrong values. Another option would be to round the Gini coefficient (and perhaps all other metrics too) to some fixed number of decimals, e.g. 5, and then use the same number of digits when testing